app/src/main/java/com/example/habithelper/utilities/HabitMultipleLinearRegression.java

```diff
@@ -0,0 +1,80 @@
+package com.example.habithelper.utilities;
+
+import org.apache.commons.math3.exception.MathIllegalArgumentException;
+import org.apache.commons.math3.linear.LUDecomposition;
+import org.apache.commons.math3.linear.QRDecomposition;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.stat.regression.AbstractMultipleLinearRegression;
+
+public class HabitMultipleLinearRegression extends AbstractMultipleLinearRegression {
+
+    private QRDecomposition qrDecomposition = null;
+
+    /**
+     * Creates an empty HabitMultipleLinearRegression instance
+     */
+    public HabitMultipleLinearRegression() {
+    }
+
+    /**
+     * Loads model habitHistoryList and moodList sample data into a RealMatrix and RealVector (respectively)
+     * and checks that their dimensions are suitable to perform MultipleLinearRegression
+     *
+     * @param moodList the numDaysTracked by 1 array representing the User's mood history
+     * @param habitHistoryList the numDaysTracked by numHabits array representing the User's habit history
+     * @throws MathIllegalArgumentException if the arrays are not the proper dimensions
+     */
+    public void newSampleData(double[] moodList, double[][] habitHistoryList) throws MathIllegalArgumentException {
```

Comment:
If the method does not override anything form the abstract class, and dince this method is doing more than loading sample Data, I may suggest to something more descriptive as loadAndValidateData() or InitAndValidateData(). Otherwise, naming aligns with overrides newXSampleData and newYSampleData